lib/components/artworks/index.js

```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+// 1. Get first layout pass of grid view so we have a total width and calculate the column width (componentDidMount?).
+// 2. Possibly do artwork column layout now, as we can do so based just on the aspect ratio, assuming the text height
+//    won't be too different between artworks.
+// 3. Get artwork heights by either:
+//    - calculating the item size upfront with aspect ratio and a static height for the text labels.
+//    - leting the artwork component do a layout pass and calculate its own height based on the column width.
+// 4. Update height of grid to encompass all items.
+
+import Relay from 'react-relay';
+import React from 'react-native';
+const { View, ScrollView, StyleSheet } = React;
+
+import Artwork from './artwork';
+import { metaphysicsURL } from '../../relay/config';
+
+/**
+ * TODOs:
+ * - currently all the code assumes column layout
+ *   - do no invert aspect ratios in row layout
+ * - deal with edge-cases when calculating in which section an artwork should go
+ *   - see ARMasonryCollectionViewLayout for details on how to deal with last works sticking out
+ *   - the calculation currently only takes into account the size of the image, not if e.g. the sale message is present
+ */
+class Artworks extends React.Component {
+  static defaultProps = {
+    sectionDirection: 'column',
+    sectionCount: 2,
+    sectionMargin: 8,
+    itemMargin: 8,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { sectionDimension: 0, artworks: this.props.artworks, page: 1 };
+
+    this.onLayout = this.onLayout.bind(this);
+    this.onScroll = this.onScroll.bind(this);
+
+    this._sentEndForContentLength = null;
+  }
+
+  fetchNextPage() {
+    const nextPage = this.state.page + 1;
+    const query = artworksQuery(this.props.relay.route.params.artistID, nextPage);
+    fetch(metaphysicsURL, { method: 'post',
```

Comment:
I just mimiced what Relay does. It also probably shows-up nicer in the networking tab of the dev tools than a GET would.